src/CachetCoreServiceProvider.php

```diff
@@ -111,14 +112,36 @@ private function configureRateLimiting(): void
     private function registerRoutes(): void
     {
         $this->callAfterResolving('router', function (Router $router, Application $application) {
-            $router->group([
-                'domain' => config('cachet.domain'),
-                'as' => 'cachet.api.',
-                'prefix' => Cachet::path().'/api',
-                'middleware' => ['cachet:api', 'throttle:cachet-api'],
-            ], function (Router $router) {
-                $this->loadRoutesFrom(__DIR__.'/../routes/api.php');
-            });
+            $apiEnabled = true;
+            $apiProtected = false;
+            try {
+                $settings = app(AppSettings::class);
+                $apiEnabled = $settings->api_enabled;
+                $apiProtected = $settings->api_protected;
+
+            } catch (MissingSettings $exception) {
+                if(! $application->runningInConsole()) {
+                    throw new \Exception("Please run `php artisan migrate` to load missing settings.");
+                }
+            } catch (\Exception $exception) {
+                // do nothing
+            }
+
+            if ($apiEnabled) {
+                $middleware = ['cachet:api', 'throttle:cachet-api'];
+                if ($apiProtected) {
+                    $middleware[] = 'auth:sanctum';
```

Comment:
This middleware is already added to the non-GET routes. I think we should consider adding a middleware that is always applied, but conditionally authorises if the setting is enabled.